### PR TITLE
 [MRG+2] Resample_img clip fix issue #1528

### LINKED ISSUE #1528: Offer clip option in resample_img
Currently using `resample_img` with `interpolation='continuous'` (default) can lead to values outside of the input range (e.g. getting negative values). `skimage` has a [`clip` option](http://scikit-image.org/docs/dev/api/skimage.transform.html?highlight=resize#skimage.transform.resize) which states:
>Whether to clip the output to the range of values of the input image. This is enabled by default, since higher order interpolation may produce values outside the given input range.

### PR DESCRIPTION
**Pull request by poquirion**
Should fix #1528

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
Looks good to me.

@cancan101 : do you want to check that this fixes your issue?

**Comment by GaelVaroquaux** [comment]
+1 for merge on my side.

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -331,6 +331,11 @@ def resample_img(img, target_affine=None, target_shape=None,
         Data ordering in output array. This function is slightly faster with
         Fortran ordering.
 
+    clip: bool, optional
+        If False (default) no clip is preformed.
```
Comment (KamalakerDadi) [inline nilearn/image/resampling.py]: preformed -> performed

File: nilearn/image/resampling.py [hunk 2]
```diff
@@ -575,6 +585,11 @@ def resample_to_img(source_img, target_img,
...
```
Comment (KamalakerDadi) [inline nilearn/image/resampling.py]: preformed -> performed

**Comment by KamalakerDadi** [comment]
travis should be fixed with this PR #1555 .

**Comment by bthirion** [comment]
LGTM.

**Comment by cancan101** [comment]
any desire to set the default to something like `None` to allow changing the default down the road from False to true?

**Comment by GaelVaroquaux** [comment]
There was something that was arguably a bug here: in the case of extrapolation (expansion of the field of view), zeros are added for padding. These zeros were being clipped out :)

**Pull request merged**